framework/src/org/checkerframework/framework/type/AnnotatedTypeMirror.java

```diff
@@ -1512,9 +1509,6 @@ void setLowerBoundField(AnnotatedTypeMirror type) {
          * @return the lower bound field.
          */
         public AnnotatedTypeMirror getLowerBoundField() {
-            if (lowerBound != null) {
```

Comment:
Thanks for this comment!
I made the "annotations" field private and added a new protected getAnnotationsField method.
There are currently no calls of that method that modify the result. I documented that such modifications might break invariants.

I added a few other cleanups in the latest commits. Please merge if you like them.